code/game/turfs/plating.dm

```diff
@@ -8,6 +8,33 @@
 	barefootstep = FOOTSTEP_HARD
 	mediumxenofootstep = FOOTSTEP_PLATING
 
+/turf/open/floor/plating/broken_states()
+	return list("platingdmg1", "platingdmg2", "platingdmg3")
+
+/turf/open/floor/plating/burnt_states()
+	return list("panelscorched")
+
+/turf/open/floor/plating/fire_act(exposed_temperature, exposed_volume)
+	if(hull_floor)
+		return
+	if(!burnt && prob(5))
+		burn_tile()
+
+/turf/open/floor/plating/welder_act(mob/living/user, obj/item/I)
+	var/obj/item/tool/weldingtool/welder = I
+
+	if(!broken && !burnt)
```

Comment:
Meant to be or? Kinda weird if not since it'd mean you can't repair only burnt or only broken floors.

Reply:
early return if they are neither burnt or broken 4head

Reply:
goober moment